Bottom: 545cac09b0c45882bc72cdcd6e5c2ee6e67ff3a2
Top:    752c61c18ce133118532dcd6167c5e46efaed624
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-21 02:21:29 +0000

HIDGL Don't bother capping hairlines



---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index c66daca..be0a6a7 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -300,9 +300,12 @@ hidgl_draw_line (int cap, double width, int x1, int y1, int x2, int y2, double s
   float deltax, deltay, length;
   float wdx, wdy;
   int circular_caps = 0;
+  int no_caps = 0;
 
-  if (width == 0.0)
+  if (width == 0.0) {
+    no_caps = 1; /* Don't bother capping hairlines */
     width = 1.0;
+  }
 
   deltax = x2 - x1;
   deltay = y2 - y1;
@@ -333,8 +336,11 @@ hidgl_draw_line (int cap, double width, int x1, int y1, int x2, int y2, double s
   switch (cap) {
     case Trace_Cap:
     case Round_Cap:
-      circular_caps = 1;
-      break;
+      if (!no_caps) {
+        circular_caps = 1;
+        break;
+      }
+      /* Fall through */
 
     case Square_Cap:
     case Beveled_Cap:
